client/client.go

```diff
@@ -129,38 +156,67 @@ func createClientWithClient(ctx context.Context, searchCli sserver1.SearchServer
 			return nil, err
 		}
 
-		indexer := libsearch.CreateSecureIndexBuilder(sha256.New, masterSecrets[i], tlfInfo.Salts, uint64(tlfInfo.Size))
-
-		dirInfo := DirectoryInfo{
-			absDir:      absDir,
-			tlfID:       tlfID,
-			indexer:     indexer,
-			pathnameKey: pathnameKey,
+		var indexers []*libsearch.SecureIndexBuilder
+		var pathnameKeys []libsearch.PathnameKeyType
+
+		// Sets up the indexers and pathname keys
+		if keyGen == libkbfs.PublicKeyGen {
+			masterSecret, err := fetchMasterSecret(directory, keyGen, lenMS)
```

Comment:
Hmm, this will fail if someone other than a writer tries to index a public directory for the first time.  Maybe that's what we want, though we could instead just use a hardcoded "secret" key.  What do you think?